vivo-project/Vitro
# Use same logic to remove blank nodes from reasoner's internal model as used in RDFService

**[VIVO GitHub issue 3884](https://github.com/vivo-project/VIVO/issues/3884)**: 

# What does this pull request do?
Refactors the logic for treating removed blank nodes as variables from RDFServiceJena to JenaModelUtils where it can be used by the JFactTBoxReasoner.  This allows ontologies removed using Add/Remove RDF to properly update the reasoner and allow related inferences to be removed.  Adds some comments and a unit test.

# How should this be tested?
- Unit test should pass.
- For additional manual confirmation, run the steps in the linked issue to confirm that the bug can no longer be reproduced.

# Interested parties
@VIVO-project/vivo-committers
